### DISCUSSION TITLE
Is there a way to reduce text2path output to a single path?

### DISCUSSION THREAD

**Discussion by gagemw** [mozman/ezdxf#442] [Q&A]
Text2path's make_paths_from_entity decomposes single line fonts into inner and outer paths, is there an elegant way to recombine those paths, or otherwise decompose text into a single path that can be used to draw line based stick text? (text with no internal area)

@mozman

**Comment by mozman** [comment]
The `Path` class deliberately does not support a `moveto` command to keep the implementation simple, therefore it can not represent multiple paths as a single object. `Matplotlib` and `PyQt` paths do support the `moveto` command and can represent multiple paths in a single object. 

The `ezdxf.path.to_matplotlib()` function combines multiple `ezdxf.path,Path` objects  into a single `Matplotlib.path.Path` object.

The `ezdxf.path.to_qpainter_path()` function combines multiple `ezdxf.path,Path` objects into a single `PyQt5.QtGui.QPainterPath` object.

**Reply by mozman** [reply]
I think I'll add the `moveto` command to the `Path` class. This will simplify and unify the processing of the `HATCH` entity that is a multi-path object and therefore needs a special treatment without the `moveto` command.

**Comment by mozman** [comment]
In the current unreleased version v0.16.4 is a new `ezdxf.path.to_multi_path()` function, which merges all given paths into a single `Path` object. But there is no separation  of letters possible, because the underlying font rendering tool Matplotlib `TextPath()` returns all given letters of a string in a single Matplotlib `Path()` object. There are tools to detect nested paths in the `ezdxf.path.nesting` module to create `HATCH` entities, but separated parts of a letter like `"i"` are also always separated `Path` objects.

Added new functions `text2path.make_path_from_str()` and `text2path.make_path_from_entity()` which return a multi-path object.
